### ISSUE TITLE
What's next?

### ISSUE THREAD

**Issue by dope** [dope/lemonade#22]
Hey,

I'm sat here wondering whats next for Lemonade. As it stands it is just a 12 column grid where you can define new columns. Do we need more than that? Do we need an optional stylesheet with a basic reset, styled buttons and typography? 

What do you want to see?

Thank you :)

**Comment by joeguilmette** [comment]
I want lemonade to be exactly what it is. A grid. There's too much shit out
there that does too much. Lemonade is perfect right now, if you want to
make a lightweight framework, great! But please don't make me fork lemonade
☺

**Comment by dope** [comment]
@joeguilmette Perfect response :)

**Comment by GabrielGil** [comment]
I agree with @joeguilmette

**Comment by Madebym** [comment]
I would leave it as it is, but one thing is most definitely missing, at least in my opinion.

Stacking columns is not enough for responsive design. Let's say that I have a four column grid for desktop. Maybe I want bit-4 to be bit-2 for tablet devices, thus giving me 2 columns in each row, because the space is definitely there for two, but not for four.

Anyway, all of the big frameworks have this, and this is the only thing missing here. It really helps with responsive design.

**Comment by Madebym** [comment]
I love Lemonade the way it is, but one thing is missing IMHO. Prefixed classes for different breakpoints, that really makes developing responsive sites much easier.

Sometimes we really need that certain columns ocuppy different widths based on a certain breakpoint. Hell, I even sometimes want to have two, or maybe even three columns in a row on small mobile breakpoint. 

This is the only thing I like about bigger frameworks, since from my experience it really helps. I just used UIKit from the big boys and it has proven invaluable.

Bottom line, it would make the grid more powerfull, yet it would still be just a little humble grid, with no extra shit and bloat like all the other grids/frameworks.

**Comment by joeguilmette** [comment]
You would just set bit-4 to 50% at the tablet breakpoint, if I understand
your issue correctly.

**Comment by Madebym** [comment]
I would just like the grid to have more options when it comes to responsive design, e.g. bits can take different form(widths) at certain breakpoints.

Yes, setting bit-4 at 50% for tablet, but I would also like the option to leave it as 25%, or even make it 100% for tablet. It is all about having more grid options :)

I would just like to have an option of alternate bit widths at certain breakpoints. Since english is not my primary language, maybe if you want you can check these two links to better understand what I want.
http://getskel.com/docs#grid-system-responsive
http://getuikit.com/docs/grid.html

**Comment by dope** [comment]
I completely get this request, I use these sort of classes on personal and work projects. It's definitely something I'd consider for a future release. What sort of naming convention would you expect?

**Comment by dope** [comment]
@Madebym So this is something I've just mocked up and tested 

```
<div class="frame">
    <div class="bit-3 bit-mobile-full">Test</div>
    <div class="bit-3 bit-mobile-full">Test</div>
    <div class="bit-3 bit-mobile-full">Test</div>
</div>

.bit-mobile-full {
    @media (max-width: $small) {
        width: 100% !important;
    }
}

.bit-mobile-half {
    @media (max-width: $small) {
        width: 50% !important;
    }
}

.bit-mobile-third {
    @media (max-width: $small) {
        width: 33.3333% !important;
    }
}

.bit-mobile-quarter {
    @media (max-width: $small) {
        width: 25% !important;
    }
}

.bit-mobile-fifth {
    @media (max-width: $small) {
        width: 20% !important;
    }
}
```

**Thoughts?**

**Comment by Madebym** [comment]
@dope 

Yes, that code you provided is exactly what I was thinking. As for the naming convention you could either use full names, like small, medium, large, etc. or shorter prefixes, like .sm, .md, .lg.

At least those are the two naming conventions I came across with different frameworks. Although I prefer the full one, not the bootstrap .sm type.

Also, in UIKit framework, when you use the non-prefixed column, in Lemonade's case it would be just bit-X, it affects all device widths and  grid columns stay side by side, without stacking, thus giving one more option.

**Comment by vkeusebio** [comment]
@dope I agree with @joeguilmette first response.

Lemonade is good as it is right now and I don't think there is a need to further extend it to something else.

**Comment by dope** [comment]
@vkeusebio Thanks for the feedback. I think I'll close this issue for now.

**Comment by vkeusebio** [comment]
@dope I think we could actually extend lemonade to support nested blocks, IF it doesn't support it currently.
